src/test/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/repository/OpenApiDocsTest.kt

```diff
@@ -0,0 +1,52 @@
+package uk.gov.justice.digital.hmpps.prisontonhs.repository
+
+import io.swagger.v3.parser.OpenAPIV3Parser
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.boot.web.server.LocalServerPort
+import org.springframework.http.MediaType
+import uk.gov.justice.digital.hmpps.prisontonhs.integration.IntegrationTest
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+class OpenApiDocsTest : IntegrationTest() {
+
+  @LocalServerPort
+  private var port: Int = 0
+
+  @Test
+  fun `open api docs are available`() {
+    webTestClient.get()
+      .uri("/swagger-ui/?configUrl=/v3/api-docs")
+      .accept(MediaType.APPLICATION_JSON)
+      .exchange()
+      .expectStatus().isOk
+  }
+
+  @Test
+  fun `the open api json reports no errors`() {
+    webTestClient.get()
+      .uri("/v3/api-docs")
+      .accept(MediaType.APPLICATION_JSON)
+      .exchange()
+      .expectStatus().isOk
+      .expectBody().jsonPath("messages").doesNotExist()
+  }
+
+  @Test
+  fun `the open api json contains the version number`() {
+    webTestClient.get()
+      .uri("/v3/api-docs")
+      .accept(MediaType.APPLICATION_JSON)
+      .exchange()
+      .expectStatus().isOk
+      .expectBody().jsonPath("info.version").isEqualTo(DateTimeFormatter.ISO_DATE.format(LocalDate.now()))
+  }
+
+  @Test
+  fun `the open api json is nearly valid and contains documentation`() {
+    val result = OpenAPIV3Parser().readLocation("http://localhost:$port/v3/api-docs", null, null)
+    assertThat(result.openAPI.paths).isNotEmpty
+    assertThat(result.messages).hasSizeLessThan(2)
```

Comment:
there is still one error here to do with response `Page<NhsPrisoner>` but ran out of time to get it working